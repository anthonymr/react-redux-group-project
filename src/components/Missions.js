import { useSelector } from 'react-redux';
import Mission from './Mission';
import style from '../styles/Missions.module.css';

function Missions() {
  const { missions } = useSelector((state) => state.missions);

  return (
    <section>
      <table className={style.missionstable}>
        <thead className={style.missionstablehead}>
          <tr className={style.missionstablerow}>
            <th scope="col" style={{ width: '10%' }}>Mission</th>
            <th scope="col" style={{ width: '60%' }}>Description </th>
            <th scope="col" style={{ width: '15%' }}>Status</th>
            <th scope="col" style={{ width: '15%' }}>Join</th>
          </tr>
        </thead>
        { missions.map((mission) => (
          <tbody className={style.tablebody} key={mission.id}>
            <Mission
              name={mission.name}
              description={mission.description}
              id={mission.id}
              reserved={mission?.reserved}
            />
          </tbody>
        ))}
      </table>
    </section>
  );
}

export default Missions;
