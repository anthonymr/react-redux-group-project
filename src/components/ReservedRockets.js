import { useSelector } from 'react-redux';
import styles from '../styles/ReservedRockets.module.css';

function ReservedRockets() {
  const { rockets } = useSelector((status) => status.rockets);

  const bookedRockets = rockets.filter((rocket) => rocket.reserved);

  return (
    <section className={styles.reservedContainer}>
      <h2>My Rockets</h2>
      <ul>
        {
            bookedRockets.map((rocket) => (
              <li key={rocket.id}>{rocket.name}</li>
            ))
        }
      </ul>
    </section>
  );
}

export default ReservedRockets;
