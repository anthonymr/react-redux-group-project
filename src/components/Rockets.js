import { useSelector } from 'react-redux';
import Rocket from './Rocket';

function Rockets() {
  const { rockets } = useSelector((state) => state.rockets);

  return (
    <section>
      <ul>
        {
          rockets.map((rocket) => (
            <li key={rocket.id}>
              <Rocket flickrImages={rocket.flickr_images} name={rocket.name} description={rocket.description} />
            </li>
          ))
        }
      </ul>
    </section>
  );
}

export default Rockets;
