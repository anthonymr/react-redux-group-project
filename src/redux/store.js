import { configureStore } from '@reduxjs/toolkit';
import missionsReducer from './missions/missionSlice';

const store = configureStore({
  reducer: {
    missions: missionsReducer,
  },
});

export default store;
