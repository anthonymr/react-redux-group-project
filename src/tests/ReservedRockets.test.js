import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../redux/store';
import Missions from '../components/Missions';
import Rockets from '../components/Rockets';
import ReservedRockets from '../components/ReservedRockets';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Missions />
        <Rockets />
        <ReservedRockets />
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
