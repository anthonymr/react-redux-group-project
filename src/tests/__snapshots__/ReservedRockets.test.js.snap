// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <section>
    <table>
      <thead>
        <tr>
          <th
            scope="col"
            style={
              Object {
                "width": "10%",
              }
            }
          >
            Mission
          </th>
          <th
            scope="col"
            style={
              Object {
                "width": "60%",
              }
            }
          >
            Description 
          </th>
          <th
            scope="col"
            style={
              Object {
                "width": "15%",
              }
            }
          >
            Status
          </th>
          <th
            scope="col"
            style={
              Object {
                "width": "15%",
              }
            }
          >
            Join
          </th>
        </tr>
      </thead>
    </table>
  </section>,
  <section
    className="rocketsContainer"
  >
    <ul
      className="rocketList"
    />
  </section>,
  <section
    className="reservedContainer"
  >
    <h2>
      My Rockets
    </h2>
    <ul />
  </section>,
]
`;
